website/client/components/userMenu/profile.vue

```diff
@@ -866,6 +898,40 @@ export default {
       this.userReceivingGems = this.user;
       this.$root.$emit('bv::show::modal', 'send-gems');
     },
+    adminRevokeChat () {
+      if (!this.hero.flags) {
+        this.hero.flags = {
+          chatRevoked: true,
+        };
+      }
+      this.hero.flags.chatRevoked = true;
+
+      this.$store.dispatch('hall:updateHero', { heroDetails: this.hero });
+    },
+    adminReinstateChat () {
+      if (!this.hero.flags) {
+        this.hero.flags = {
```

Comment:
both here and in the previous fiunction can you just do

```
if (!this.hero.flags) {
        this.hero.flags = {}
}
```

since the value is set to true anyway in the next line

Reply:
Sounds good, I'll fix this up